internal/api/courses/assignments/list_test.go

```diff
@@ -10,25 +10,49 @@ import (
 )
 
 func TestList(test *testing.T) {
+	users := db.MustGetServerUsers()
+
 	testCases := []struct {
-		course string
+		email     string
+		permError bool
+		course    string
 	}{
-		// Empty
-		{"course-without-source"},
-		// One Assignment
-		{"course101-with-zero-limit"},
-		// Multiple Assignments
-		{"course-languages"},
+		// Invalid permissions.
+		{"server-user@test.com", true, "course-without-source"},
+		{"server-creator@test.com", true, "course-without-source"},
+
+		// Valid permissions, Empty
```

Comment:
No need to repeat "Valid Permissions" on each comment.
Just make one comment denoting this "section" of tests.